File: src/main/java/com/aries/tutorial/batch/ApplicationServiceBatch.java
package com.aries.tutorial.batch;

import com.aries.extension.data.BatchData;
import com.aries.extension.data.batch.ApplicationServiceData;
import com.aries.extension.handler.BatchHandler;
import com.aries.extension.util.PropertyUtil;
import com.aries.extension.util.LogUtil;

public class ApplicationServiceBatch implements BatchHandler {
    @Override
    public boolean preHandle(long batchTime) {
        // TODO: Adding pre-processing code
        return true;
    }

    @Override
    public void process(BatchData[] batchData) {
        LogUtil.info("[ApplicationServiceBatch] - " +
                PropertyUtil.getValue("application_service_batch", "subject", "Unknown subject"));

        for(int i = 0; i < batchData.length; i++) {
            ApplicationServiceData data = (ApplicationServiceData) batchData[i];

            StringBuilder sb = new StringBuilder();
            sb.append("Domain ID : " + data.domainId + ", ");
            sb.append("Domain Name : " + data.domainName + ", ");
            sb.append("Instance Name : " + data.instanceName + ", ");
            sb.append("Application Name : " + data.applicationName + ", ");
            sb.append("Call Count : " + data.callCount + ", ");
            sb.append("Failure Count : " + data.failureCount);

            LogUtil.info(sb.toString());
        }
    }
}

File: src/main/kotlin/com/aries/tutorial2/batch/ApplicationServiceBatch.kt
package com.aries.tutorial2.batch

import com.aries.extension.data.BatchData
import com.aries.extension.data.batch.ApplicationServiceData
import com.aries.extension.handler.BatchHandler
import com.aries.extension.util.LogUtil
import com.aries.extension.util.PropertyUtil

class ApplicationServiceBatch : BatchHandler {
    override fun preHandle(batchTime: Long): Boolean {
        // TODO: Adding pre-processing code
        return true
    }

    override fun process(batchData: Array<BatchData>) {
        LogUtil.info("[ApplicationServiceBatch] - " + PropertyUtil.getValue("application_service_batch", "subject", "Unknown subject"))

        for (d in batchData) {
            val data = d as ApplicationServiceData

            LogUtil.info("Domain ID : " + data.domainId)
            LogUtil.info("Domain Name : " + data.domainName)
            LogUtil.info("Instance Name : " + data.instanceName)
            LogUtil.info("Application Name : " + data.applicationName)
            LogUtil.info("Call Count : " + data.callCount)
            LogUtil.info("Failure Count : " + data.failureCount + "\n")
        }
    }
}

